lib/model/internal_link.dart

```diff
@@ -292,3 +303,28 @@ List<int>? _parseDmOperand(String operand) {
     return null;
   }
 }
+
+/// An operand value of "is" operator.
+///
+/// See also:
+///   - https://zulip.com/api/construct-narrow
+///   - https://zulip.com/help/search-for-messages#search-your-important-messages
+///   - https://zulip.com/help/search-for-messages#search-by-message-status
+@JsonEnum(alwaysCreate: true)
+enum IsOperand {
+  dm,
+  private, // This is deprecated in FL 177 and equivalent to [dm].
```

Comment:
```suggestion
  dm, // TODO(server-7) new in FL 177
  private, // TODO(server-7) deprecated in FL 177, equivalent to [dm]
```

The `TODO(server-N)` format will help us spot in the future that we can drop `private`.

For the newly-introduced value, having a comment puts us on guard to not try to use it with old servers. Then we need a TODO-server comment to remind us to delete that warning comment… so it may as well be one line that does both.

Reply:
Somewhat related: how do we deal with deprecated API values? Do we still need to support them post server-7, until they are fully removed in a future version?

Reply:
When it comes to links, we'll want to accept the old forms forever (as will web). People will have put links to Zulip in all sorts of external places we can't update — for example in GitHub comments, for Zulip users who use GitHub like we do — and so we should keep those links working.

But once we stop supporting the old server versions that lack the new forms, we can stop ever generating the old forms. So for example ApiNarrowDm and its two subclasses will simplify back down to just ApiNarrowDm, and the only wrinkle remaining for the long term will be that we'll accept `pm-with` in links and translate that to ApiNarrowDm. We'll handle IsOperand.private similarly.